### DISCUSSION TITLE
Why do we convert to ServiceBusReceivedMessage ourselves and not using the OOTB conversion?

### DISCUSSION THREAD

**Discussion by renebik** [arcus-azure/arcus.templates#905] [General]
In the ServiceBus.Queue and ServiceBus.Topic templates we have an explicit method `ConvertToServiceBusMessage` to do the conversion to `ServiceBusReceivedMessage`. Why not using the `ServiceBusReceivedMessage` directly like so: 
```C#
        public async Task Run(
            [ServiceBusTrigger("orders", Connection = "ServiceBusConnectionString")] ServiceBusReceivedMessage message,
            FunctionContext executionContext)
```
[(source)](https://github.com/arcus-azure/arcus.templates/blob/main/src/Arcus.Templates.AzureFunctions.ServiceBus.Queue/OrderFunction.cs#L45)

**Comment by stijnmoreels** [comment]
Bc at the time of creating the template, Microsoft didn't supported this way of working and we had no choice than doing it ourselvse.
The new version of the templates will be using the message type directly.
